### PR TITLE
Full Re-Work on the UI

### PR DESCRIPTION
**Pull request by OcelotWalrus**
There is too much to say, just test the stuff yourself.

I think, instead of making this game a text RPG, it could be good if we make it more an ASCII RPG, for example add ascii art for every map zone and print it to the player:
some stuff like this:
image

And also add ASCII art to each items.

Other stuff that would be good is critical hit chance, an other attributes added only to weapons, the more you have critical hit chance, the more you have a small chance to deal a critical heat to the enemy, which will result in damaging your enemy with twice the damage you can technically deal without critical hit.

Battle risk, a small formula that calculates the player winning chance in percents and print it at the very first of the fight, when the player decide to try to run away, fight or use an item.

Day - Night cycles, which result in player loosing agility/dodging chance, and maybe more. Mybe if it is night, you can find monsters the randomly spawns. In the `zone.yaml` file, each zone will specify monsters to randomly spawns if it is night time.

Maybe instead of showing health has `<current health>/<max health>`, it could be a sort of a progress bar, but the HP/Max HP will still be shown after the bar.

Add NPC, that have kind of the same work as the enemy work. Add the quest book, similar to the diary, where the player can see all his done and in progress quests.

Add gold, a player stat. Maybe will be used with NPC.

Add a new thing (don't know how to call it), that will work like items, when specified on the map, it will print "You cannot continue that way, you don't have the necessary key". You'll need the right item to continue that way.

Add a new attribute that will make the game treat the coordinates as 'a village', which will be shown on the map. There will be a `village.yaml` file where all data about the village is written, as an example:
```
"""village.yaml"""

Kathallion Village:
  name: "Kathallion Village"
  description: "A small village in the Plains of Kathallion"
  contents:
    hostel: # if there is no hostel, the prompt will look the same as always. If there is a hostel, there will be a different prompt that you can access by running a command (for example 'V')
      name: "The Kathallion Hostel"
      cost value: 1.1 # if the beef is 3 gold, in this village the beef will be 3.3 gold.
      sleep gold: 7 # the gold you pay to stay the night here and stay away from monsters
      items to buy: # items you can buy here, items will have a new attribute, cost in gold
        - Beef
        - Prokchop
        - Cooked Chicken
        ...
      drinks to buy: # like items but you don't actually get items, you just 'drink' to get health back. There will be a drinks.yaml file
        - Beer
        ...
```

### TODO
- [x] Make Main Menu
- [x] Change Player UI
- [x] Enhance Player UI
- [x] Add Village Features
- [x] Add Gold
- [x] Add Drinks
- [x] Add Battle Stuff (battle risks, health bar, critical hit)
- [x] Add More ASCII Art Everywhere
- [x] Add NPC
- [x] Add Key Stuff
- [x] Add Python Modules Requirements To README.md

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,7 @@ EXAMPLE*
 __pycache__
 __pycache__/**
 **/save_*
+plugins/
```
Comment (lumbar527) [inline .gitignore:7]:
```suggestion
**/~0*
plugins/
```

**Comment by OcelotWalrus** [comment]
I think what I'm gonna work on now is adding more ASCII stuff everywhere.

**Comment by lumbar527** [comment]
I can't test yet; need to get some of the modules.

**Comment by OcelotWalrus** [comment]
> I can't test yet; need to get some of the modules.

Yes.

File: main.py [hunk 2]
```diff
@@ -427,140 +968,92 @@ def run(play):
...
-            if player["held leggings"] == " ":
-                print("You are currently holding no leggings")
-            else:
-                print("You are wearing a/an " + COLOR_RED + player["held leggings"] + COLOR_RESET_ALL)
-            print(" ")
-            which_item = input("You have these items in your inventory: " + str(player["inventory"]) + " ")
+            if player["held item"] != " ":
+                print("HELD WEAPON: " + COLOR_RED + COLOR_STYLE_BRIGHT + player["held item"] + COLOR_RESET_ALL)
+            if player["held chestplate"] != " ":
+                print("WORE CHESTPLATE: " + COLOR_RED + COLOR_STYLE_BRIGHT + player["held chestplate"] + COLOR_RESET_ALL)
```
Comment (lumbar527) [inline main.py:983]:
```suggestion
                print("WORN CHESTPLATE: " + COLOR_RED + COLOR_STYLE_BRIGHT + player["held chestplate"] + COLOR_RESET_ALL)
```

File: main.py [hunk 3]
```diff
@@ -427,140 +968,92 @@ def run(play):
...
+            if player["held leggings"] != " ":
+                print("WORE LEGGINGS: " + COLOR_RED + COLOR_STYLE_BRIGHT + player["held leggings"] + COLOR_RESET_ALL)
```
Comment (lumbar527) [inline main.py:985]:
```suggestion
                print("WORN LEGGINGS: " + COLOR_RED + COLOR_STYLE_BRIGHT + player["held leggings"] + COLOR_RESET_ALL)
```

File: main.py [hunk 4]
```diff
@@ -427,140 +968,92 @@ def run(play):
...
+            if player["held boots"] != " ":
+                print("WORE BOOTS: " + COLOR_RED + COLOR_STYLE_BRIGHT + player["held boots"] + COLOR_RESET_ALL)
```
Comment (lumbar527) [inline main.py:987]:
```suggestion
                print("WORN BOOTS: " + COLOR_RED + COLOR_STYLE_BRIGHT + player["held boots"] + COLOR_RESET_ALL)
```

**Review by lumbar527 — COMMENTED** [review]
I still haven't tested

File: preferences.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+latest preset:
+  plugin: Cromha Stuff
+  save: Cromha MCcloud
+  type: plugin
```
Comment (lumbar527) [inline preferences.yaml:4]: These will have to be reset.

File: test.py
Comment (lumbar527) [inline test.py:1]: What is this for?

File: imgs/Title2.txt
Comment (lumbar527) [inline imgs/Title2.txt:1]: What does this do?

(I need to test stuff soon)

Comment (OcelotWalrus) [reply imgs/Title2.txt:1]: Example:
image
I'm really almost finish, just have some stuff to finish (battle & npc UI) and the PR will be ready.

File: preferences.yaml (hunk 5 above)
Comment (OcelotWalrus) [reply preferences.yaml:4]: Yes.

Comment (OcelotWalrus) [reply preferences.yaml:4]: I can't get the `preferences.yaml` be in the .gitignore. Even if I add
```
preferences.yaml
**/preferences.yaml
```
It don't get added.

File: test.py
Comment (OcelotWalrus) [reply test.py:1]: Temporary file :)

File: imgs/Title2.txt
Comment (lumbar527) [reply imgs/Title2.txt:1]: Well, I still need to test.

**Comment by OcelotWalrus** [comment]
I've finished everything about the UI re-work and other new features.

There is only villages plus hostels ( will be easy, just have to copy NPC code) and the 'Key' stuff ( an item needed for the player to enter a specific location).

**Comment by OcelotWalrus** [comment]
About ASCII: enemies & npc have some ascii art to illustrate them and they both supports colors (I recommend to work on adding colors to them.) Items will also have thumbnails.

I will try to make/find more simple ASCII art for every existing enemies and items.

It works by adding a file with this syntax: `imgs/<item/npc/enemy name>.txt`. Plugins also needs to have a `imgs/` folder inside of them.

**Comment by OcelotWalrus** [comment]
Should I also add with hostels, something like churches? Maybe they could sell special templar weaponry? Or maybe also make so that you can pray and it will make something?

**Comment by lumbar527** [comment]
I would rather not.

**Comment by OcelotWalrus** [comment]
> I would rather not.

Sure!

**Comment by OcelotWalrus** [comment]
I've potentially finished village features. By the way, now all items needs a `gold` attributes, that defines the price of that item. The `gold` attributes can be decimal like `3.45`. I let you decide a price for each item that are currently in the vanilla game.

I think you can test the PR now.

**Comment by OcelotWalrus** [comment]
I have one question:

In this PR, I added a day/night cycle, so sleeping in an hostel really does something. Should I make that when it's night, generic monsters have a chance to spawns? Or maybe a new list of monsters that is made especially for day/night cycles.

So in that 'night list' there will be slimes, zombies... Or maybe not zombies?

**Comment by OcelotWalrus** [comment]
Okay, so I close the PR?

**Comment by lumbar527** [comment]
Yeah, I can totally put it in the organization. What do you want it to be called?

**Comment by OcelotWalrus** [comment]
I think I'll keep it as a branch for now while I think about a name, but anyway, thanks you a lot!

**Comment by lumbar527** [comment]
OK, do you mean you want to work more on your fork and think of a name before I add it to the organization?

**Comment by OcelotWalrus** [comment]
Yes.

**Comment by lumbar527** [comment]
Just tell me when you're ready and I'll put it in the organization.

**Comment by OcelotWalrus** [comment]
Yes.

**Comment by OcelotWalrus** [comment]
Have you any idea of a name?

**Comment by OcelotWalrus** [comment]
Maybe Bane Of Wargs?

**Comment by lumbar527** [comment]
Sure!

**Comment by OcelotWalrus** [comment]
Thanks!

**Comment by OcelotWalrus** [comment]
Actually it seems I can create repositories in the organization myself.

**Comment by lumbar527** [comment]
Great! I'll close this now.

**Comment by OcelotWalrus** [comment]
Sure

**Pull request closed without merging**